sources/obstacles/circular_obstacle.cpp

```diff
@@ -0,0 +1,90 @@
+#include "obstacle.h"
+#include "circular_obstacle.h"
+#include "rectangular_obstacle.h"
+#include "../struct/vector_2d.h"
+#include <cmath>
+#include <typeinfo>
+
+namespace kraken
+{
+    CircularObstacle::CircularObstacle(const Vector2D &pos, float radius) :
+            Obstacle(pos), radius_(radius), squared_radius_(radius * radius)
+    {
+    }
+
+    bool CircularObstacle::isInObstacle(const Vector2D &pos) const
+    {
+        return pos.squaredDistance(rotation_center_) <= squared_radius_;
+    }
+
+    float CircularObstacle::squaredDistance(const Vector2D &pos) const
+    {
+        // TODO : is "distance" really necessary ? (it uses a std::sqrt computation)
+        float out = std::max(0.f, pos.distance(rotation_center_) - radius_);
```

Comment:
We should consider implementing an approximation of the euclidean distance that does not uses sqrt and add an entry in the config for this.